crates/ruff_server/src/server.rs

```diff
@@ -46,9 +45,7 @@ impl Server {
             .workspace_folders
             .map(|folders| folders.into_iter().map(|folder| folder.uri).collect())
             .or_else(|| init_params.root_uri.map(|u| vec![u]))
-            .ok_or_else(|| {
-                anyhow!("No workspace or root URI was given in the LSP initialization parameters. The server cannot start.")
-            })?;
+            .unwrap_or_default();
```

Comment:
If I understand this correctly, then the workspaces will be empty, right? Should we instead default to using the current working directory? In `ruff-lsp`, we default to using the current working directory as the workspace with the default settings. You might be more aware of how that will affect other parts of the code.

Reply:
I think we should use the current working directory as the default workspace if not provided.

Reply:
You're right, we should try to use the current working directory. Let me know if that works!